nemo_skills/inference/server/code_execution_model.py

```diff
@@ -220,6 +220,17 @@ def _recover_from_error(self, request, new_output, executor):
 
         return most_common
 
+    def _handle_stop_words(self, output: str):
+        """
+        OpenAI chat API remove stop word from the output, so we need to add it back
+        to enable code execution.
+        """
+        if not isinstance(self.model, OpenAIModel):
+            return output
+        if output.find(CODE_SEPARATORS[0]) > output.find(CODE_SEPARATORS[-1]):
```

Comment:
Can you please check if we can do it in the same way as for vllm? The api should return which stop words caused it to finish and we can directly add that instead of hardcoding it for our separators

Reply:
Unfortunately I did not find any way to get the exact stop word. Looks like this is only possible with legacy completions API.